src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/UseIntegralOrStringArgumentForIndexers.cs

```diff
@@ -31,16 +31,17 @@ public sealed class UseIntegralOrStringArgumentForIndexersAnalyzer : DiagnosticA
                                                                              isPortedFxCopRule: true,
                                                                              isDataflowRule: false);
 
-        private static readonly SpecialType[] s_allowedSpecialTypes = new[] {
-                        SpecialType.System_String,
-                        SpecialType.System_Int16,
-                        SpecialType.System_Int32,
-                        SpecialType.System_Int64,
-                        SpecialType.System_Object,
-                        SpecialType.System_UInt16,
-                        SpecialType.System_UInt32,
-                        SpecialType.System_UInt64
-                        };
+        private static readonly ImmutableHashSet<SpecialType> s_allowedSpecialTypes =
```

Comment:
📝 It's a near certainty that this change went the opposite direction of what was intended. `ImmutableHashSet<T>` has poor locality and lookup performance (every element is stored in a different object in memory), while the one case where linear searches excel is small arrays of primitive types.

Reply:
Thanks @sharwell. @Evangelink would you be able to fix it up with a follow-up PR with a PERF comment on the field on the same lines?

Reply:
@sharwell Thanks that's interesting! Is it a remark specific to the immutables or is it true also for regular collections? My point is if I have a small collection is it better to have array or `HashSet` to do a `Contains`?

Reply:
See #3666